### ISSUE TITLE
Custom directives bug

### ISSUE THREAD

**Issue by zheeeng** [vuejs/babel-plugin-transform-vue-jsx#69]
From README.md

In this usage: 
```
v-clickoutside={{ expression: 'dismiss' }}
```

mydirective print out the `binding` is a structure likes that:

```
def: Object,
modifiers: Object,
name: "clickoutside",
value: {
  expression: "dismiss"
}
```

The passed object `{ expression: 'dismiss' }` is assigned to `binding.value` not the `binding` object.

**Comment by LinusBorg** [comment]
I think you are misunderstanding the [API ](https://github.com/vuejs/babel-plugin-transform-vue-jsx#vue-directives)(but maybe it's me, I'm not 100% familiar with the implementation yet).

I think you can't pass in an object with properties for the [hook arguments](https://vuejs.org/v2/guide/custom-directive.html#Directive-Hook-Arguments), you can only pass properties for the [VNodeDirective interface](https://vuejs.org/v2/guide/custom-directive.html#Directive-Hook-Arguments) - this 

Also, this syntax requires a value property to be present, but you don't haveone in your example, so the object itself is assumed to *be* the value.

**Comment by jacelynfish** [comment]
I've committed a pull request trying to solve this problem by delegating the attributes of an object-form custom directive value onto the `binding` object. Please [check it out](https://github.com/vuejs/babel-plugin-transform-vue-jsx/pull/78).

**Comment by nickmessing** [comment]
@jacelynfish, can you please do some tests

**Comment by jacelynfish** [comment]
@nickmessing I have tested it running the mocha testing cases provided.

**Comment by nickmessing** [comment]
Wait for #78 to be merged and 3.4.4 to be released.
